src/components/AddBook.js

```diff
@@ -1,11 +1,59 @@
-import React from 'react';
-
-const AddBook = () => (
-  <form>
-    <input type="text" name="Title" placeholder="Title" />
-    <input type="text" name="Author" placeholder="Author" />
-    <input type="button" value="Add" />
-  </form>
-);
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+
+const AddBook = (props) => {
+  const [input, setInput] = useState({
+    title: '',
+    author: '',
+  });
+
+  const onChange = (e) => {
+    setInput({
+      ...input,
+      [e.target.name]: e.target.value,
+    });
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (input.title.trim() && input.author.trim()) {
+      props.propsToAddBooks(input.title, input.author);
+      setInput({
+        title: '',
+        author: '',
+      });
+    }
+  };
```

Comment:
- [ ] For this milestone, you are required to generate a unique ID while saving the book data from the inputs, and then dispatches the addBook action with the object as its argument.